.github/workflows/program-nft-packs.yml

```diff
@@ -0,0 +1,51 @@
+name: Program NFT Packs
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+env:
+  CARGO_TERM_COLOR: always
+  SOLANA_VERSION: 1.8.5
+  RUST_TOOLCHAIN: stable
+
+jobs:
+  build-and-test-nft-packs:
+    runs-on: ubuntu-latest
+    env:
+      cache_id: program-nft-packs 
+
+    steps:
+      # Setup Deps
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/install-linux-build-deps
+      - uses: ./.github/actions/install-solana
+        with: 
+          solana_version: ${{ env.SOLANA_VERSION }}
+      - uses: ./.github/actions/install-rust
+        with: 
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+
+      # Restore Cache from previous build/test
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ./rust/target
+          key: ${{ env.cache_id }}-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-${{ env.RUSTC_HASH }}
+
+      # Build deps
+      - uses: ./.github/actions/build-metaplex
+      - uses: ./.github/actions/build-token-metadata
```

Comment:
Token metadata should already build when you build metaplex, so no need to build it again.

Reply:
When we build metaplex program it doesn't produce necessary *.so file so we need to build token-metadata
It happens because programs are building as dynamic libraries